src/wayland/shm/mod.rs

```diff
@@ -366,6 +378,147 @@ pub fn has_alpha(format: wl_shm::Format) -> bool {
     )
 }
 
+macro_rules! shm_format_table {
+    (
+        $(
+            $fourcc: ident => $shm: ident
+        ),* $(,)?
+    ) => {
+        /// Convert from a [`Fourcc`](backend::allocator::Fourcc) format to a wl_shm format
+        pub const fn fourcc_to_shm_format(value: $crate::backend::allocator::Fourcc) -> Option<$crate::reexports::wayland_server::protocol::wl_shm::Format> {
+            match value {
+                $(
+                    $crate::backend::allocator::Fourcc::$fourcc => Some($crate::reexports::wayland_server::protocol::wl_shm::Format::$shm),
+                )*
+                _ => None,
+            }
+        }
+
+        /// Convert from a wl_shm format to a [`Fourcc`](backend::allocator::Fourcc) format
+        pub const fn shm_format_to_fourcc(value: $crate::reexports::wayland_server::protocol::wl_shm::Format) -> Option<$crate::backend::allocator::Fourcc> {
+            match value {
+                $(
+                    $crate::reexports::wayland_server::protocol::wl_shm::Format::$shm => Some($crate::backend::allocator::Fourcc::$fourcc),
+                )*
+                _ => None,
+            }
+        }
+    }
+}
+
+shm_format_table! {
```

Comment:
Given that most formats directly map we could extend the macro to define a drm format without specifying the shm variant and just pass the drm ident as the shm ident. But I am not sure this is really worth extending the macro.

Reply:
Honestly I like the explicit variant and its not like we will be extending that list every other day.